librabbitmq/amqp_openssl_hostname_validation.c

```diff
@@ -70,8 +71,11 @@ static amqp_hostname_validation_result amqp_matches_common_name(
   if (common_name_asn1 == NULL) {
     return AMQP_HVR_ERROR;
   }
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
   common_name_str = (char *)ASN1_STRING_data(common_name_asn1);
-
+#else
+  common_name_str = (char *)ASN1_STRING_get0_data(common_name_asn1);
```

Comment:
Casting away const is not necessary, and should be avoided.

Reply:
Cast removed.

Reply:
I revert last change. Travis build fail without cast.